src/plugin/dl/dlwrappers.cpp

```diff
@@ -71,29 +72,41 @@ getRpathRunPath(void *caller, string *rpathStr, string *runpathStr)
   ASSERT_NOT_NULL(strtab);
 
   if (rpath != NULL) {
-    *rpathStr = strtab + rpath->d_un.d_val;
-    *rpathStr = dmtcp::Util::replace(*rpathStr, "$ORIGIN", dirname);
+    Util::replace(rpathStr, &strtab[rpath->d_un.d_val], "$ORIGIN", dirname);
```

Comment:
'man ld.so' says that `${ORIGIN}` is also valid.  We should also test for that.

Also, we're not supporting `$LIB`, `${LIB}`, `$PLATFORM`, or `${PLATFORM`.
We should at least provide a FIXME comment here.
Ooptionally, we could add code to test for this and issue a warning/error.